src/client/services/socketIo.service.js

```diff
@@ -0,0 +1,18 @@
+import io from 'socket.io-client';
+import { store } from 'resources/store';
+import { UPDATE_USER } from 'resources/user/user.actions';
+
+const socket = io(window.config.webSocketUrl, { transports: ['websocket'], query: { token: window.token } });
+
+socket.on('connect', () => {
+  console.log('WS connected'); //eslint-disable-line
+  socket.emit('subscribe', window.user._id);
+
+  socket.on('user:updated', (user) => {
```

Comment:
The handler logic should be placed outside `socketIo.service.js` itself. 

`socketIo.service` should expose `on`, `emit`, `subscribe` and `unsubscribe` methods that should be used in handler files

**Example usage:**

```
import socketService from 'services/socketIo.service.js';

socketService.on('user:updated', user => store.dispatch(...));
```

**Another use-case for the future usage with subscribe call**

```
componentWillMount() {
   socketService.subscribe(`chat-1234`);
   socketService.on('message:created', ...);
}

componentWillUnmount() {
  socketService.unsubscribe(`chat-1234`);
}
```